src/renderers/shared/reconciler/ReactCompositeComponent.js

```diff
@@ -498,6 +498,7 @@ var ReactCompositeComponentMixin = {
     var inst = this._instance;
     var childContext = inst.getChildContext && inst.getChildContext();
     if (childContext) {
+      ReactCurrentOwner.processingChildContext = true;
```

Comment:
Probably don't want to attach this to ReactCurrentOwner, since it has nothing to do with owners.  But conceptually, yes, we want to set some sort of value when we start a `getChildContext`.

I think it probably makes sense to use the new devtools api here.  You can emit an event when you start/end getting the child context, and keep the state in a devtool instead of tying it into the core.  Devtool api details: https://github.com/facebook/react/pull/5590/files